datafusion/core/src/datasource/provider.rs

```diff
@@ -161,20 +161,73 @@ pub trait TableProvider: Sync + Send {
     /// Specify if DataFusion should provide filter expressions to the
     /// TableProvider to apply *during* the scan.
     ///
+    /// Some TableProviders can evaluate filters more efficiently than the
+    /// `Filter` operator in DataFusion, for example by using an index.
+    /// 
     /// The return value must have one element for each filter expression passed
     /// in. The value of each element indicates if the TableProvider can apply
-    /// that particular filter during the scan.
+    /// that particular filter during the scan. The position in the return value
+    /// Vec corresponds to the expression in the `filters` input. 
+    /// 
+    /// Here is an example of how this can be done:
+    /// 
+    /// ```rust
+    /// use datafusion::error::{Result, DataFusionError};
+    /// use datafusion_expr::{Expr, TableProviderFilterPushDown};
+    /// 
+    /// // Override the supports_filters_pushdown to evaluate which expressions 
+    /// // to accept as pushdown predicates.
+    /// fn supports_filters_pushdown(&self, filters: &[&Expr]) -> Result<Vec<TableProviderFilterPushDown>> {
     ///
-    /// Some TableProviders can evaluate filters more efficiently than the
-    /// `Filter` operator in DataFusion, for example by using an index.
+    ///     let mut result_vec: Vec<TableProviderFilterPushDown> = Vec::with_capacity(&filters.len());
     ///
-    /// By default, returns [`Unsupported`] for all filters, meaning no filters
+    ///     // Process each filter
+    ///     for i in 0..filters.len() {
+    ///         let expr = *filters[i];
+    ///
+    ///         let table_provider_filter_pushdown = match expr {
+    ///             // This example only supports a between expr with a single column name.
+    ///             Expr::Between(between_expr) => {
+    ///                 between_expr.expr.try_into_col()
+    ///                     .map(|column| {
+    ///                         if column.name = "c1".to_string() {
+    ///                             TableProviderFilterPushDown::Exact
+    ///                         } else {
+    ///                             TableProviderFilterPushDown::Unsupported
+    ///                         }
+    ///                     })
+    ///                     // If there is no column in the expr set the filter to unsupported.
+    ///                     .unwrap_or(TableProviderFilterPushDown::Unsupported);
```

Comment:
Need to remove the `;` here so that this match arm is returning something and not returning the unit type `()`.

```suggestion
    ///                     .unwrap_or(TableProviderFilterPushDown::Unsupported)
```